### ISSUE TITLE
"Type[Exception]" is not iterable error when handling exception

### ISSUE THREAD

**Issue by roganov** [microsoft/pyright#816]
**Describe the bug**
pyright is reporting a weird error `"Type[Exception]" is not iterable` when handling exceptions.

**To Reproduce**
```python
from typing import Type


class Exceptions:
    exc: Type[Exception] = Exception


try:
    1/0
except Exceptions().exc:
    print('exc')
```

**Expected behavior**
No errors

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is fixed in Pyright 1.1.52, which I just published. It will be included in the next published version of Pylance.
